cpu/atmega1284p/cpu.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2016 Robert Hartung <hartung@ibr.cs.tu-bs.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_atmega1284p
+ * @{
+ *
+ * @file
+ * @brief       Implementation of the CPU initialization
+ *
+ * @author      Robert Hartung <hartung@ibr.cs.tu-bs.de>
+ * @}
+ */
+
+#include "cpu.h"
+#include "periph/init.h"
+#include <avr/wdt.h>
+
+#if (__AVR_LIBC_VERSION__ >= 10700UL)
+/* The proper way to set the signature is */
+#include <avr/signature.h>
+#else
+
+/* signature API not available before avr-lib-1.7.0. Do it manually.*/
```

Comment:
out of curiosity, what is this signature for?

Reply:
If you're on a programmer (e.g. the inga_tool) it has to know the signature of the MCU (the model). This just ensures it's available.

Reply:
Cool! Thanks :-)